docs/spec.md

```diff
@@ -764,8 +764,12 @@ Within each process group in the StableHLO grid, splits the values of the
 between the processes, concatenates the scattered parts along `concat_dimension`
 and produces a `result` tensor.
 
-The operation splits the StableHLO grid into process groups using the
-`cross_replica(replica_groups)` strategy.
+The operation splits the StableHLO grid into `process_groups` as follows:
+
+  * `channel_id <= 0`,
+    `cross_replica(replica_groups)`.
+  * `channel_id > 0`,
+    `cross_partition(replica_groups)`.
```

Comment:
Shouldn't the constraint C6 also needs to change ?

(C6) `size(replica_groups)` depends on the process grouping strategy:
    * If `cross_replica`, `num_replicas`.
    * If `cross_partition`, `num_partition`.

Reply:
Nice catch!